webapp/packages/core-utils/src/MetadataMap.test.ts

```diff
@@ -0,0 +1,237 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { z } from 'zod';
+
+import { MetadataMap } from './MetadataMap';
+
+describe('MetadataMap', () => {
+  it('should create an empty map', () => {
+    const map = new MetadataMap<number, string>();
+
+    expect(map.size).toBe(0);
+  });
+
+  it('should sync items', () => {
+    const map = new MetadataMap<number, string>();
+    const emptyMap = new MetadataMap<number, string>();
+
+    const data: [number, string][] = [
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+      [Infinity, 'infinity'],
+      [NaN, 'nan'],
+    ];
+
+    map.sync(data);
+    emptyMap.sync([]);
+
+    data.forEach(([key, value]) => {
+      expect(map.get(key)).toBe(value);
+    });
+
+    expect(emptyMap.size).toBe(0);
+  });
+
+  it('should set items', () => {
+    const map = new MetadataMap<number, string>();
+    map.set(1, 'one');
+    map.set(2, 'two');
+    map.set(3, 'three');
+    map.set(Infinity, 'infinity');
+    map.set(NaN, 'nan');
+
+    expect(map.get(1)).toBe('one');
+    expect(map.get(2)).toBe('two');
+    expect(map.get(3)).toBe('three');
+    expect(map.get(Infinity)).toBe('infinity');
+    expect(map.get(NaN)).toBe('nan');
+  });
+
+  it('should throw an error on invalidate with no default value getter', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    try {
+      map.get(1, undefined, z.object({}));
+      expect(true).toBeFalsy(); // should not be called
+    } catch (e) {
+      expect(e).toBeInstanceOf(Error);
+    }
+
+    try {
+      map.get(1);
+      expect(true).toBeFalsy(); // should not be called
+    } catch (e) {
+      expect(e).toBeInstanceOf(Error);
+    }
+  });
+
+  it('should get function default value', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    expect(JSON.stringify(map.get(1, key => `default ${key}`, z.object({})))).toStrictEqual(JSON.stringify('default 1'));
```

Comment:
This should not work this way; also, why do you pass an empty zod object?
In this case, `map.get(1, key => `default ${key}`)` should return `one`
also, I can't understand the purpose of using JSON.stringify

Reply:
I see. Please rename the test case because it tests fallback to the default value in case when schema validation failed

Reply:
but still why you use JSON.stringify in such many places?